### PR TITLE
:wrench: Add option to bypass ControlNet input image crop

### LINKED ISSUE #2148: [Bug]: ControlNet and Inpaint Mask not working for some preprocessors in A1111
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### What happened?

Using Automatic1111 and latest ControlNet, some of the preprocessors like LineArt, Canny, Softedge are not working. This happens specifically when "Only Masked" is selected. In some cases, the controlnet output is a black box, black box with slight horizontal lines, or a cropped out version of the input picture. 
one (2)
two (1)
three (1)

Steps: 20, Sampler: Euler a, CFG scale: 3, Seed: 999, Size: 512x512, Model hash: 4c86efd062, Model: SDv1-5, VAE hash: c6a580b13a, VAE: vae-ft-mse-840000-ema-pruned.vae.pt, Denoising strength: 1, Init image hash: 26a8e14cb29034d412d74767b77559e6, Mask blur: 4, ControlNet 0: "Module: lineart_standard (from white bg & black line), Model: control_v11p_sd15_lineart [43d4be0d], Weight: 1, Resize Mode: Resize and Fill, Low Vram: False, Processor Res: 512, Guidance Start: 0, Guidance End: 1, Pixel Perfect: False, Control Mode: Balanced", Version: v1.6.0

EDIT: The SD Model used doesn't impact anything. I've tried the base 1.5, the 1.5 inpainting, and a bunch of other checkpoints with no impact on the CN output. Just to add. 

### Steps to reproduce the problem

1. Go to A1111 web ui
2. Go to img2Img tab
3. Go to inpaint upload tab
4. Upload image and mask of your choice
5. Set Inpaint area as "Only Masked"
6. Open ControlNet dropdown and add a picture. 
7. Select One of these control types from the list (Canny, Depth, Softedge, LineArt). I have tried all these. 
8. Select "Resize and Fill" as a resize mode. 
9. Generate

### What should have happened?

ControlNet should output properly sized images after processing which then gets fed to the geneartion of the masked part of the image. 
It's not resizing and filling properly in this instance. In past experiences, it just outputs black square.

### Commit where the problem happens

webui: version: [v1.6.0](https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/5ef669de080814067961f28357256e8fe27544f4)
controlnet: tried using v1.1.410 1.1.408, 1.1.313, 1.1.312,1.1.237, 1.1.134,1.1.230,1.1.221, and 1.1.179 to mention a few.

### What browsers do you use to access the UI ?

_No response_

### Command Line Arguments

```Shell
No
```

### List of enabled extensions

extensions

### Console logs

```Shell
ControlNet - INFO - Loading model: control_v11p_sd15_lineart [43d4be0d]
 ControlNet - INFO - Loaded state_dict from [/workspace/sd/stable-diffusion-webui/extensions/sd-webui-controlnet/models/control_v11p_sd15_lineart.pth]
 ControlNet - INFO - controlnet_default_config
 ControlNet - INFO - ControlNet model control_v11p_sd15_lineart [43d4be0d] loaded.
 ControlNet - INFO - Loading preprocessor: lineart_standard
 ControlNet - INFO - preprocessor resolution = 512
 ControlNet - INFO - ControlNet Hooked - Time = 10.276776790618896
100%|███████████████████████████████████████████| 16/16 [00:02<00:00,  7.04it/s]
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by huchenlei**
Closes #2148.

This PR adds a new option in ControlNet settings to allow user disable crop of input image when inpaint area is set to "Only masked".

When this option is selected, the image crop will still be active when user invokes upscale script, which also uses inpaint area = "Only maksed" to crop tiles off the input image.

### CODE REVIEW

**Comment by Gushousekai195** [comment]
Awesome! You think this could be a setting we could enable on one controlnet unit and leave disabled on another?

**Comment by Gushousekai195** [comment]
> Closes #2148.
> 
> This PR adds a new option in ControlNet settings to allow user disable crop of input image when inpaint area is set to "Only masked".
> 
> When this option is selected, the image crop will still be active when user invokes upscale script, which also uses inpaint area = "Only maksed" to crop tiles off the input image.

Awesome! You think this could be a setting we could enable on one controlnet unit and disable on another?

**Comment by 2blackbar** [comment]
Id rather have that on controlnet panel itself

**Comment by Gushousekai195** [comment]
> Id rather have that on controlnet panel itself

What he said

**Comment by huchenlei** [comment]
@2blackbar @Gushousekai195 Done in https://github.com/Mikubill/sd-webui-controlnet/pull/2377.
